src/BizHawk.Client.EmuHawk/tools/Lua/LuaConsole.cs

```diff
@@ -209,6 +209,7 @@ public override void Restart()
 
 			LuaFileList newScripts = new(LuaImp?.ScriptList, onChanged: SessionChangedCallback);
 			LuaFunctionList registeredFuncList = new(onChanged: UpdateRegisteredFunctionsDialog);
+			(LuaImp as Win32LuaLibraries)?.Close();

```

Comment:
> Causes open forms to be closed on core restart

Not sure I like this. Ext. tool forms remain open, though they get `Restart` called...

Reply:
I can see why keeping lua forms open may be desirable, but as it is it's not functional.

Any event handlers of the forms will still be attached to the old `LuaImp` and even the services of the old core (memory etc.). And there's no way for the restarted script in the new `LuaImp` to even interact with any forms/controls that remained open, the handles will be lost and the new `FormsLuaLibrary` isn't able to map them anyway.

Reply:
For event handlers, could we add a level of indirection?

What do you mean by "the handles will be lost"? If you mean `FormsLuaLibrary._luaForms` is garbage collected, that can be changed.

Reply:
Variables, including global ones, won't persist through the `Restart`. So scripts would have to be specifically written to persist the form handles somewhere and retrieve them after restart.

`Restart` will instantiate a new `FormsLuaLibrary` that doesn't know about the previous `_luaForms`, so currently even if the restarted script knows the handle of the open form or its controls, functions like `forms.settext` won't work. This would have to be changed somehow.

Also, I don't think there's any non-hacky way for a script to check if an old handle is still valid

Reply:
Oh, sorry, I didn't even think of that.  `client.reboot_core` already works fine with keeping forms open, none of the issues I mentioned apply to that.

This change doesn't affect that case, `Restart()` bails out early if  `luaLibsImpl.IsRebootingCore` is set.